src/serializers/type_serializers/union.rs

```diff
@@ -78,13 +78,14 @@ impl TypeSerializer for UnionSerializer {
         // try the serializers in left to right order with error_on fallback=true
         let mut new_extra = extra.clone();
         new_extra.check = SerCheck::Strict;
+        let mut errors: Vec<PyErr> = Vec::new();
```

Comment:
I suggest using `SmallVec` similar to in the validation case (so we don't allocate for small unions). Same in each of the below.